tests/common/test_visualization.py

```diff
@@ -0,0 +1,163 @@
+
+@pytest.mark.parametrize(
+    "file_name, nodes_count, edges_count, relationship_node_keys",
+    [
+        (
+            "SPDXJSONExample-v2.3.spdx.json",
+            22,
+            22,
+            ["SPDXRef-Package_DYNAMIC_LINK", "SPDXRef-JenaLib_CONTAINS"],
+        ),
+        (
+            "SPDXJSONExample-v2.2.spdx.json",
+            20,
+            19,
+            ["SPDXRef-Package_DYNAMIC_LINK", "SPDXRef-JenaLib_CONTAINS"],
+        ),
+        (
+            "SPDXRdfExample-v2.3.spdx.rdf.xml",
+            22,
+            22,
+            ["SPDXRef-Package_DYNAMIC_LINK", "SPDXRef-JenaLib_CONTAINS"],
+        ),
+        (
+            "SPDXRdfExample-v2.2.spdx.rdf.xml",
+            20,
+            17,
+            ["SPDXRef-Package_DYNAMIC_LINK", "SPDXRef-JenaLib_CONTAINS"],
+        ),
+        (
+            "SPDXTagExample-v2.3.spdx",
+            22,
+            22,
+            ["SPDXRef-Package_DYNAMIC_LINK", "SPDXRef-JenaLib_CONTAINS"],
+        ),
+    ],
+)
+def test_generate_graph_from_spdx(
+    file_name: str,
+    nodes_count: int,
+    edges_count: int,
+    relationship_node_keys: List[str],
+) -> None:
+    document = parse_file(str(Path(__file__).resolve().parent.parent / "spdx" / "data" / "formats" / file_name))
+    graph = generate_graph_from_spdx(document)
+
+    assert document.creation_info.spdx_id in graph.nodes()
+
+    assert graph.number_of_nodes() == nodes_count
+    assert graph.number_of_edges() == edges_count
+    assert "SPDXRef-DOCUMENT_DESCRIBES" in graph.nodes()
+    for relationship_node_key in relationship_node_keys:
+        assert relationship_node_key in graph.nodes()
+
+
+def test_complete_connected_graph() -> None:
+    document = _create_minimal_document()
+
+    graph = generate_graph_from_spdx(document)
+
+    TestCase().assertCountEqual(
+        graph.nodes(),
+        [
+            "SPDXRef-DOCUMENT",
+            "SPDXRef-Package-A",
+            "SPDXRef-Package-B",
+            "SPDXRef-File",
+            "SPDXRef-DOCUMENT_DESCRIBES",
+            "SPDXRef-Package-A_CONTAINS",
+            "SPDXRef-Package-B_CONTAINS",
+        ],
+    )
+    TestCase().assertCountEqual(
+        graph.edges(),
+        [
+            ("SPDXRef-DOCUMENT", "SPDXRef-DOCUMENT_DESCRIBES"),
+            ("SPDXRef-DOCUMENT_DESCRIBES", "SPDXRef-Package-A"),
+            ("SPDXRef-DOCUMENT_DESCRIBES", "SPDXRef-Package-B"),
+            ("SPDXRef-Package-A", "SPDXRef-Package-A_CONTAINS"),
+            ("SPDXRef-Package-A_CONTAINS", "SPDXRef-File"),
+            ("SPDXRef-Package-B", "SPDXRef-Package-B_CONTAINS"),
+            ("SPDXRef-Package-B_CONTAINS", "SPDXRef-File"),
+        ],
+    )
+
+
+def test_complete_unconnected_graph() -> None:
+    document = _create_minimal_document()
+    document.packages += [
+        Package(
+            spdx_id="SPDXRef-Package-C",
+            name="Package without connection to document",
+            download_location="https://download.location.com",
```

Comment:
Why do you provide a download location here?

Reply:
Basically to make this package valid, which was needed in an earlier development. I think I will just use a `package_fixture` here.